### PR TITLE
Kv list keys

### PR DESCRIPTION
**Pull request by brandynbennett**
Add a KV function to get all the values from a KV bucket.

Moved the inbox/nuid creation into the general `Util` module because it's useful for more than just the `PullConsumer` module.

### CODE REVIEW

File: lib/jetstream/api/kv.ex [hunk 1]
```diff
@@ -151,6 +152,56 @@ defmodule Jetstream.API.KV do
...
 
+  @doc """
+  Get all the non-deleted values for a Key Value Bucket
+
+  ## Examples
+
+      iex>%{"key1" => "value1} = Jetstream.API.KV.all_values(:gnat, "my_bucket")
+  """
+  @spec all_values(conn :: Gnat.t(), bucket_name :: binary()) :: %{}
+  def all_values(conn, bucket_name) do
```
Comment (mmmries) [inline lib/jetstream/api/kv.ex:163]: The name `all_values` reminds me of `Map.values` from the standard library, but that function only returns a list of values without their associated keys. 

Does NATS documentation have a term for getting the non-deleted entries from a key-value bucket? Maybe this is already named after that documentation?

Comment (mmmries) [reply lib/jetstream/api/kv.ex:163]: What about `contents(conn, bucket)`? That way you're getting the non-deleted contents of a bucket?

Comment (brandynbennett) [reply lib/jetstream/api/kv.ex:163]: That's a good point. Thanks for pointing that out. I like `contents` better.

Comment (brandynbennett) [reply lib/jetstream/api/kv.ex]: @mmmries updated here: https://github.com/mmmries/jetstream/pull/64/commits/3f65e5be4920f97c5a9ee86c6547dd1cbc32dc13

Comment (mkaput) [reply lib/jetstream/api/kv.ex]: This looks like https://hexdocs.pm/elixir/Map.html#to_list/1. I have also thought about word `entries`.

File: mix.exs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 defmodule Jetstream.MixProject do
   use Mix.Project
 
-  @version "0.0.2"
+  @version "0.0.3"
```
Comment (mkaput) [inline mix.exs:4]: I think bumping version number is out of scope of this PR 😃

File: lib/jetstream/api/kv.ex [hunk 3]
```diff
@@ -151,6 +152,56 @@ defmodule Jetstream.API.KV do
...
 
+  @doc """
+  Get all the non-deleted key-value pairs for a Bucket
+
+  ## Examples
+
+      iex>%{"key1" => "value1} = Jetstream.API.KV.contents(:gnat, "my_bucket")
+  """
+  @spec contents(conn :: Gnat.t(), bucket_name :: binary()) :: %{}
+  def contents(conn, bucket_name) do
```
Comment (mkaput) [inline lib/jetstream/api/kv.ex:163]: This raises when Gnat.sub or Consumer.create fails. Should this be a bang function?

```suggestion
  @spec contents!(conn :: Gnat.t(), bucket_name :: binary()) :: %{}
  def contents!(conn, bucket_name) do
```

File: mix.exs (hunk 2 above)
Comment (brandynbennett) [reply mix.exs]: https://github.com/mmmries/jetstream/pull/64/commits/2ce7e538ec2cf5668289824b375193caa74ca090

File: lib/jetstream/api/kv.ex (hunk 3 above)
Comment (brandynbennett) [reply lib/jetstream/api/kv.ex:163]: I think this does point out that the `contents` function should return an error tuple as well if either of the underlying functions fail. I'll refactor to include that

Comment (brandynbennett) [reply lib/jetstream/api/kv.ex]: https://github.com/mmmries/jetstream/pull/64/commits/0af6c3ce3ac3bbef8a9019b23061ea47d0f0cdc1

**Review by mmmries — APPROVED** [review]
:shipit: Looks great! ❤️ 💛 💙 💚 💜

**Pull request merged**